java-rx-smgrsdk/src/main/java/io/vantiq/svcconnector/SvcConnectorServer.java

```diff
@@ -1,49 +1,69 @@
+ * verticles that handle incoming requests from the vantiq server and process them. It is necessitated by the dance with
+ * configuring and starting Vert.x
+ * <p/>
  * Copyright (c) 2023 Vantiq, Inc.
- * <p>
+ * <p/>
  * All rights reserved.
  */
 @Slf4j
 public class SvcConnectorServer {
     static Vertx vertx;
     
-    public static Vertx getVertx() {
-        if (vertx == null) {
-            throw new IllegalStateException("Vertx has not been initialized");
-        }
-        return vertx;
-    }
-    
     public void start(SvcConnectorConfig config) {
         System.setProperty("vertx.logger-delegate-factory-class-name", "io.vertx.core.logging.SLF4JLogDelegateFactory");
         
         // convert the config to a JsonObject -- suitable for verticle deployment
         JsonObject verticleConfig = new JsonObject().put("storageManagerClassName", config.getStorageManagerClassName());
         Runtime runtime = Runtime.getRuntime();
-        log.info("Detected {} processors, starting {} instances of the request processing verticle",
-                runtime.availableProcessors(), 4*runtime.availableProcessors());
+        // ToDo: will want to adjust some vertx options here (blocked thread checking etc.) when in debug mode.
+        VertxOptions options = new VertxOptions();
+        vertx = Vertx.vertx(options);
+        
+        // define a couple of codecs, so we can send messages between verticles
+        vertx.eventBus().registerDefaultCodec(SvcConnSvrMessage.class, new SvcConnSvcMsgCodec());
+        vertx.eventBus().registerDefaultCodec(StorageManagerError.class, new StorageManagerErrorCodec());
+
+        log.info("Detected {} processors, starting {} instances of the websocket processing verticle and " +
+                        "{} instances of the storage manager verticle",
+                runtime.availableProcessors(), runtime.availableProcessors(), 4*runtime.availableProcessors());
+        
+        // start a web socket verticle for each of the processors. it is the case that once a connection is established
+        // all subsequent requests are handled by the same vertx event loop.
         DeploymentOptions deployOptions = new DeploymentOptions()
-                .setInstances(4*runtime.availableProcessors())
+                .setInstances(runtime.availableProcessors())
                 .setConfig(verticleConfig);
-        vertx = Vertx.vertx();
-        Supplier<Verticle> verticleSupplier = RequestProcessingVerticle::new;
-        vertx.deployVerticle(verticleSupplier, deployOptions).onFailure(t -> {
+        vertx.deployVerticle(WebSocketRequestVerticle::new, deployOptions).onFailure(t -> {
             System.err.println("Failed to deploy main verticle: " + t.getMessage());
             vertx.close();
+            System.exit(1);
+        });
+        
+        // start 4x the number of processors of the storage manager verticle
```

Comment:
would the number of storage manager verticles something that we would want to eventually be able to configure?

Reply:
yes, it should be configurable. I think I will enter an issue to track it. Need to look into a mechanism to specify config files much like the Vantiq server's startup.config